### PR TITLE
[Pod] Add example of a complex dependency

### PR DESCRIPTION
**Pull request by shepting**
The Braintree dependency shows examples of:
  - nested `.framework` inside of a pod
  - `genrule` using `lipo` to remove architectures
  - dylib (`CardinalMobile`)
  - multiple `prebuilt_cxx_library` with `.a` internals as dependencies
  - localization inside a library
  - `Info.plist` substitutions
  - `apple_bundle` wrapped in `apple_resource` to get included in final bundle
  - multiple location bundles (UI and code-based)
  - `genrule` with changes based on `build_type`

Screen Shot 2020-05-15 at 1 45 30 PM

Also ensured that tests run:
Screen Shot 2020-05-15 at 2 11 14 PM

## Tested by:
- [x] Built and ran in Xcode (`make project`)
- [x] Built and ran with Buck CLI (`make debug`)

## Please Review
@bachand @qyang-nj @xianwen @dfed

### CODE REVIEW

File: App/BUCK [hunk 1]
```diff
@@ -62,6 +62,7 @@ apple_library(
     ],
     tests = app_tests,
     deps = [
+        "//Pods:Braintree",
```
Comment (shepting) [inline App/BUCK:65]: Actually require in the app.

File: Config/configs.bzl [hunk 2]
```diff
@@ -131,3 +131,23 @@ def message_binary_configs(name):
...
+        bundle_identifier):
+    return {
+        "EXECUTABLE_NAME": executable_name,
+        "PRODUCT_NAME": product_name,
+        "PRODUCT_BUNDLE_IDENTIFIER": bundle_identifier,
+    }
+
+def build_type():
+    # the default value is in .buckconfig and can be overriden by --config
+    return native.read_config("build", "type", "Debug")
```
Comment (shepting) [inline Config/configs.bzl:153]: This is a key change that allows rules to change their behavior based on the `build.type` config option.

File: Podfile [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
+  pod 'Braintree/Apple-Pay', '4.28.0'
+  pod 'Braintree/PaymentFlow', '4.28.0'
   pod 'Bugsnag'
-  pod 'PromiseKit/CorePromise', '6.1.1'
+  pod 'CryptoSwift'
+  pod 'Nimble', '8.0.2'
   pod 'PromiseKit/CoreLocation', '6.1.1'
+  pod 'PromiseKit/CorePromise', '6.1.1'
   pod 'Quick', '2.1.0'
-  pod 'Nimble', '8.0.2'
```
Comment (shepting) [inline Podfile:13]: Sorted. Added `Braintree`.

File: Podfile [hunk 4]
```diff
@@ -1,14 +1,18 @@
 # Set the platform globally
-platform :ios, '10.0'
+platform :ios, '12.0'
```
Comment (shepting) [inline Podfile:2]: Keep our versions consistent.

File: Pods/BUCK [hunk 5]
```diff
@@ -86,4 +87,178 @@ apple_third_party_lib(
...
+
+genrule(
+    name = "CardinalMobileFramework",
+    srcs = glob([
+        "Braintree/Frameworks/CardinalMobile.framework/**",
+    ]),
+    bash = """
+        cp -L -R ${SRCDIR}/Braintree/Frameworks/CardinalMobile.framework ${OUT}
+    """ + ("" if build_type() == "Debug" else """
+        lipo ${OUT}/CardinalMobile -remove i386 -remove x86_64 -output ${OUT}/CardinalMobile
```
Comment (shepting) [inline Pods/BUCK:123]: This is a key example for folks to see how `build_type` can be used to allow building with different architectures based on Debug or Release.

File: App/ViewController.swift [hunk 6]
```diff
@@ -1,17 +1,23 @@
...
 import SwiftAndObjc
 import SwiftReliesOnCXX
 import SwiftWithAssets
 import SwiftWithMLModel
 import SwiftWithPrecompiledDependency
 import YetAnotherSwiftModule
-import UIKit
+
+// Pods
+import PromiseKit
```
Comment (qyang-nj) [inline App/ViewController.swift:17]: 👍🏻

**Comment by qyang-nj** [comment]
Actually, is there any module depends on `//Pods:Braintree`?

File: Pods/BUCK (hunk 5 above)
Comment (Lcsmarcal) [reply Pods/BUCK:123]: That is pretty good!

**Pull request merged**